### DISCUSSION TITLE
Using mobx-state-tree $nonEmptyObject typescript error?

### DISCUSSION THREAD

**Discussion by cmdcolin** [mobxjs/mobx-state-tree#1699] [Q&A] [answered]
With this particular setup

```

export const Hello = types.model("Hello", {
  name: types.string,
});
export const HelloWithSnapshotProcessor = types.snapshotProcessor(Hello, {
  postProcessor: (snap) => {
    const { name, ...rest } = snap;
    if (!name) {
      return rest;
    }

    return snap;
  },
});
```

If I try to compile this in typescript I get 

```
src/index.ts:6:14 - error TS4023: Exported variable 'HelloWithSnapshotProcessor' has or is using name '$nonEmptyObject' from external module "/home/cdiesh/msttest2/node_modules/mobx-state-tree/dist/types/complex-types/model" but cannot be named.

```

Made a reproducible example here
https://github.com/cmdcolin/mst-nonempty-object-issue

**Comment by EmilTholin** [comment]
Hi @cmdcolin!

The error disappears if you remove `"declaration": true` from your `tsconfig.json`. Have you tried any of the solutions outlined [here](https://github.com/microsoft/TypeScript/issues/5711)?

**Reply by cmdcolin** [reply]
Thanks for the reply. I guess that is true, but I think I want to have declaration files emitted e.g. I want to get .d.ts files? I did end up at that thread but then I wasn't really sure what to import. The suggestion is to

"If you add an import of ClassA to ModuleA.ts, the error should go away."

But in this case I wasn't sure what I should import to make the error go away

**Reply by EmilTholin** [reply]
Yeah, I'm not suggesting you should remove `"declaration": true` entirely, just that that setting seem to be what is causing the error.

I'm not exactly sure what the underlying issue really is, and if it's an issue with MST that we can fix. I have next to no understanding of this part of the codebase, sadly.

**Reply by cmdcolin** [reply]
no worries..thanks a bunch for checking this out though. I only just ran into it trying to upgrade to the latest mobx-state-tree@5 but i am not sure if the problem is isolated to the latest release

I think mobx-state-tree 3.14.1 does not see this issue, but 3.17.3 does have it, as well as the latest mobx-state-tree@5.0.1

**Comment by cmdcolin** [comment, marked as answer]
Applying a really helpful suggestion from https://github.com/mobxjs/mobx-state-tree/issues/1524 we can get

```
export const Hello = types.model("Hello", {
  name: types.string,
});
export const HelloWithSnapshotProcessor = types.snapshotProcessor(Hello, {
  postProcessor: (snap) => {
    const { name, ...rest } = snap as Omit<typeof snap, symbol>;
    if (!name) {
      return rest;
    }

    return snap;
  },
});
```
